Added trimming (read ends) feature from FASTP to ivar pipeline

TFerrel8 opened the pull request:
Added fixed start/end trim feature for each read.
Lines 15,16, 79, 80, and 98 relate to this new feature

TFerrel8 committed:
Add files via upload

Added fixed start/end trim feature for each read.
Lines 15,16, 79, 80, and 98 relate to this new feature

khyox reviewed (changes requested):
Many thanks! You don't need to create an additional file (`ivar_updated.py`) for this. You better directly work on `ivar.py` so that we can see the additions/improvements there. Ultimately, we don't want two versions of the ivar pipeline but just one that we can tailor to the needs using options/flags. Also, git will take care of the versions, so another reason for avoiding new files.

TFerrel8 commented on mappgene/subscripts/ivar_updated.py:
```diff
@@ -0,0 +1,183 @@
+    variant_frequency = params['variant_frequency']
+    read_cutoff_bp = params['read_cutoff_bp']
+    thread_usage = params['threads'] # new
+    trim_front_tail = params['trim_f_t] #new
```
line 16 needs a " ' " at the end of trim_f_t.

TFerrel8 committed:
Rename ivar.py to ivar_.py

Remove notes from ivar and added a " ' " to the end of trim_f_t.

khyox reviewed (changes requested):
Thanks! We still have the same problem, two different files for the pipeline. We don't need to have different names for the different versions of the code, since git manages that. So, the idea is to have your changes directly in `ivar.py` so that anyone can easily check the differences between versions and follow the evolution of the code over time (creating additional files disrupts that process). Many thanks.

TFerrel8 committed:
Added filtering step to ivar pipeline

Adds filtering step based on fastp to ivar pipeline
The filtering step trims the beginning and end of the read (globally)
